cwltool/argparser.py

```diff
@@ -120,6 +120,34 @@ def arg_parser():  # type: () -> argparse.ArgumentParser
                         type=float,
                         default=20)
 
+    provgroup = parser.add_argument_group("Options for recording provenance "
+        "information of the execution")
+    provgroup.add_argument("--provenance",
+                        help="Save provenance to specified folder as a Research Object that capture and aggregate workflow execution and data products.",
+                        type=Text)
+
+    provgroup.add_argument("--enable-user-provenance", default=False, action="store_true",
+                        help="Record user account info as part of provenance.",
+                        dest="user_provenance")
+    provgroup.add_argument("--disable-user-provenance", default=False, action="store_false",
+                        help="Do not record user account info in provenance.",
+                        dest="user_provenance")
+    provgroup.add_argument("--enable-host-provenance", default=False, action="store_true",
+                        help="Record host info as part of provenance.",
+                        dest="host_provenance")
+    provgroup.add_argument("--disable-host-provenance", default=False, action="store_false",
+                        help="Do not record host info in provenance.",
+                        dest="host_provenance")
+    provgroup.add_argument("--orcid",
+                        help="Record user ORCID identifier as part of provenance, e.g. https://orcid.org/0000-0002-1825-0097 or 0000-0002-1825-0097",
+                        dest="orcid",
+                        default=os.environ.get("ORCID"),
+                        type=Text)
+    provgroup.add_argument("--full-name",
+                        help="Record full name of user as part of provenance, e.g. Josiah Carberry. You may need to use shell quotes to preserve spaces.",
```

Comment:
It's alternative to `--enable-user-provenance` picking it up from the unix account.

In theory we could look up the full name from the ORCID using their [API](https://orcid.org/organizations/integrators/API) -- but without somewhere to store this would then happen each time? Also you would 'leak' a HTTPS call for every run, but perhaps that's OK.